src/main/java/xyz/nucleoid/fantasy/mixin/registry/SimpleRegistryMixin.java

```diff
@@ -50,15 +53,15 @@ public abstract class SimpleRegistryMixin<T> implements RemoveFromRegistry<T> {
             this.rawIdToEntry.set(rawId, null);
```

Comment:
This line causes the following error after quitting to title while being in a world with temporary dimensions:
```
java.lang.NullPointerException: null
	at java.base/java.util.stream.Collectors.lambda$uniqKeysMapAccumulator$1(Collectors.java:179) ~[?:?]
	at java.base/java.util.stream.ReduceOps$3ReducingSink.accept(ReduceOps.java:169) ~[?:?]
	at it.unimi.dsi.fastutil.objects.ObjectArrayList$Spliterator.forEachRemaining(ObjectArrayList.java:955) ~[fastutil-8.5.12.jar:?]
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509) ~[?:?]
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499) ~[?:?]
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921) ~[?:?]
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[?:?]
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:682) ~[?:?]
	at net.minecraft.world.dimension.DimensionOptionsRegistryHolder.<init>(DimensionOptionsRegistryHolder.java:62) ~[minecraft-merged-837649b1eb-1.20.5-rc3-net.fabricmc.yarn.1_20_5_rc3.1.20.5-rc3+build.2-v2.jar:?]
	at net.minecraft.world.level.WorldGenSettings.encode(WorldGenSettings.java:24) ~[minecraft-merged-837649b1eb-1.20.5-rc3-net.fabricmc.yarn.1_20_5_rc3.1.20.5-rc3+build.2-v2.jar:?]
	at net.minecraft.world.level.LevelProperties.updateProperties(LevelProperties.java:256) ~[minecraft-merged-837649b1eb-1.20.5-rc3-net.fabricmc.yarn.1_20_5_rc3.1.20.5-rc3+build.2-v2.jar:?]
	at net.minecraft.world.level.LevelProperties.cloneWorldNbt(LevelProperties.java:233) ~[minecraft-merged-837649b1eb-1.20.5-rc3-net.fabricmc.yarn.1_20_5_rc3.1.20.5-rc3+build.2-v2.jar:?]
	at net.minecraft.world.level.storage.LevelStorage$Session.backupLevelDataFile(LevelStorage.java:452) ~[minecraft-merged-837649b1eb-1.20.5-rc3-net.fabricmc.yarn.1_20_5_rc3.1.20.5-rc3+build.2-v2.jar:?]
	at net.minecraft.server.MinecraftServer.save(MinecraftServer.java:557) ~[minecraft-merged-837649b1eb-1.20.5-rc3-net.fabricmc.yarn.1_20_5_rc3.1.20.5-rc3+build.2-v2.jar:?]
	at net.minecraft.server.MinecraftServer.shutdown(MinecraftServer.java:614) ~[minecraft-merged-837649b1eb-1.20.5-rc3-net.fabricmc.yarn.1_20_5_rc3.1.20.5-rc3+build.2-v2.jar:?]
	at net.minecraft.server.integrated.IntegratedServer.shutdown(IntegratedServer.java:227) ~[minecraft-merged-837649b1eb-1.20.5-rc3-net.fabricmc.yarn.1_20_5_rc3.1.20.5-rc3+build.2-v2.jar:?]
	at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:729) ~[minecraft-merged-837649b1eb-1.20.5-rc3-net.fabricmc.yarn.1_20_5_rc3.1.20.5-rc3+build.2-v2.jar:?]
	at net.minecraft.server.MinecraftServer.method_29739(MinecraftServer.java:275) ~[minecraft-merged-837649b1eb-1.20.5-rc3-net.fabricmc.yarn.1_20_5_rc3.1.20.5-rc3+build.2-v2.jar:?]
	at java.base/java.lang.Thread.run(Thread.java:1583) [?:?]
```

I tested 
```java
this.rawIdToEntry.remove(rawId);
```
instead and it worked just fine.

Reply:
I just noticed a problem with `this.rawIdToEntry.remove(rawId);` that I suggested.
When re-joining the world, another exception is thrown:
```
 java.lang.IndexOutOfBoundsException: Index (19) is greater than or equal to list size (19)
```

So maybe another method should be used...

Reply:
they removed the SimpleRegistry set method, which seem to have been used as base to make that mixin, so I'll need to see what other changes were made to understand how it works now

Reply:
You are right, the SimpleRegistry::set method was renamed to SimpleRegistry::add (see https://github.com/Patbox/polymer/commit/8c95c6ad859eb2e55a712abc6980479b9938f801#diff-b60f7a60e576bc1ac5f25a8702384c70ee8d3bc27b96e6ec67e4706030f9d38a for example).
But in the annotated line, the set method of rawIdToEntry (an ObjectList) is called.

My guess is that the value was set to null to drop the reference to the entry.
The problem is that null values cause exceptions as of 1.20.5.
When removing it from the list, the error I posted above occurs. But maybe I did something wrong...

Reply:
it maybe had a reference for that position on another place, I'll check the stack trace and debug it

Reply:
Is the experimental warning supposed to show on single player?
Also, temp worlds seem to be getting saved...
![image](https://github.com/NucleoidMC/fantasy/assets/40528665/0f1f6072-c784-4319-84f7-ad62f82b345c)

The IndexOutOfBoundsException is caused by the mod trying to remove a smaller id before the bigger one, making the bigger one have an invalid pos, eg, remove id 11, remove id 12, the list had size 13, now it is 12, so error on removing id 12, image below

This may also mess things added after fantasy (by other mods) as the id won't represent the position on the list

![image](https://github.com/NucleoidMC/fantasy/assets/40528665/d679b66c-b81d-403f-882f-52fcb4296c77)
![image](https://github.com/NucleoidMC/fantasy/assets/40528665/e5d8bd8e-2ef9-4370-981c-a0081bc01dcc)

Reply:
The warning was printed when I entered one of the temporary dimensions, which has the configs of the overworld.
I used `/execute in fantasy:xxxxxxx run tp @s 0 100 0` to get there.

Also yeah, the temp worlds probably shouldn't be saved 😄 

When removing items from that list may cause problems, and null cannot be used, would it be viable if there was some kind of "magic value", such as EMPTY or something? If not, maybe a null check can be inserted before the error occurs...

Reply:
A magic value may be safer, as if a mod tries to use that it won't have the null check, and the main class has 2 methods that get the value, and 1 that returns the iterator

But if using a magic value it can't break Minecraft :thinking:

Reply:
I also got the experimental warning loading a world I never used a fantasy world on...

Reply:
nvm magic value doesn't seem really possible
mojang not accepting null anymore is making this harder than it could have been